src/mrack/utils.py

```diff
@@ -198,13 +263,15 @@ def get_shortname(hostname):
 
 def get_username(host, meta_host, config):
     """Find username from sources db/metadata/config."""
-    username = host.username or meta_host.get("username")
-    default_user = get_config_value(config["users"], meta_host["os"])
-
+    provider = host.provider.name
+    dict_key = meta_host["os"]
```

Comment:
I think we do not need this part and use the actual `host.provider.name` and `meta_host["os"]` in the function call. This would help me personally to not confuse and see more transparently what are the things we are passing to this function. As this change does not change the functionality i will leave it to you whether you change it or not.